Ingress/controllers/gce/backends.go

```diff
@@ -40,7 +40,17 @@ func portKey(port int64) string {
 }
 
 func beName(port int64) string {
-	return fmt.Sprintf("%v-%d", backendPrefix, port)
+	// TODO: Pipe the clusterName through, for now it saves code churn to just
+	// grab it globally, especially since we haven't decided how to handle
+	// namespace conflicts in the Ubernetes context.
+	if *clusterName == "" {
+		return fmt.Sprintf("%v-%d", backendPrefix, port)
+	}
+	taggedName := fmt.Sprintf("%v-%d%v%v", backendPrefix, port, clusterNameDelimiter, *clusterName)
+	if len(taggedName) > nameLenLimit {
+		taggedName = taggedName[:nameLenLimit]
+	}
+	return taggedName
```

Comment:
I'm currently taking the simpler approach of asking users to modify the rc so it has a unique name if they want to avoid collisions. I'm probably going to acquire a uid from the master before it leaves beta.

Changed gce-cluster-name to cluster-uid